File: backend/app/routers/tourism.py
from fastapi import APIRouter, HTTPException
from ..models.schemas import TourismQueryRequest, TourismQueryResponse, PlanRequest, PlanResponse, WeatherData, ForecastDay
from ..agents.tourism_agent import TourismAgent
from ..utils.exceptions import PlaceNotFoundException

router = APIRouter()
agent = TourismAgent()

@router.post("/query", response_model=TourismQueryResponse)
async def query_tourism(request: TourismQueryRequest):
    try:
        response = await agent.run(request)
        return response
    except PlaceNotFoundException as e:
        raise HTTPException(status_code=404, detail=str(e))
    except Exception as e:
        return TourismQueryResponse(
            place=request.place or "",
            used_agents=[],
            weather_summary=None,
            places=None,
            message="An unexpected error occurred. Please try again later.",
            error="SERVER_ERROR"
        )
@router.post("/plan", response_model=PlanResponse)
async def plan_trip(request: PlanRequest):
    try:
        response = await agent.plan(request)
        return response
    except PlaceNotFoundException as e:
        raise HTTPException(status_code=404, detail=str(e))
    except Exception:
        return PlanResponse(
            place=request.place or "",
            used_agents=[],
            weather=None,
            places=None,
            message="An unexpected error occurred. Please try again later.",
            error="SERVER_ERROR"
        )

File: frontend/src/types/tourism.ts
export interface TourismQueryResponse {
  place: string;
  used_agents: string[];
  weather_summary?: string | null;
  places?: string[] | null;
  message: string;
  error?: string | null;
}

export interface ForecastDay {
  date: string;
  temp_max: number;
  temp_min: number;
  rain_prob?: number;
}

export interface WeatherData {
  current_temperature: number;
  current_rain_chance: number;
  summary: string;
  forecast: ForecastDay[];
}

export interface PlanResponse {
  place: string;
  used_agents: string[];
  weather?: WeatherData | null;
  places?: string[] | null;
  message: string;
  error?: string | null;
}
